chef/cookbooks/cinder/recipes/volume.rb

```diff
@@ -213,9 +213,17 @@ def make_volume(node, backend_id, volume)
         if ceph_servers.length > 0
           include_ceph_recipe = true
         else
-          message = "Ceph was not deployed with Crowbar yet!"
-          Chef::Log.fatal(message)
-          raise message
+          # If we don't have any osd from our query, it could be because the
+          # osd nodes temporarily lost the role (while rebooting, for instance).
+          # So check if the ceph setup was done once already, to decide whether
+          # to fail or just to emit a warning.
+          if File.exists?("/etc/ceph/ceph.client.admin.keyring")
```

Comment:
Following check should be more appropriate, because of external ceph configs
```
check_ceph = Mixlib::ShellOut.new("ceph -k #{volume['rbd']['admin_keyring']} \
                                                                  -c #{volume['rbd']['config_file']} -s |\ 
                                                                  grep -q -e 'HEALTH_[OK|WARN]'")
check_ceph.run_command
if check_ceph.exitstatus == 0
...
```